pkg/git/repository.go

```diff
@@ -97,6 +98,59 @@ func (r *Repository) WriteFile(src io.Reader, dst string) error {
 	return err
 }
 
+// Returns the single directory under the environments folder for a given repo
+// Returns an error if there was a problem in doing so (including if more than one folder found)
+// string return type for ease of mocking, callers would use .Name() anyway
+func (r *Repository) GetUniqueEnvironmentFolder() (string, error) {
+	topLevelDirs, err := r.DirectoriesUnderPath(r.repoPath())
+	var topLevelDirsNoDotGit []os.FileInfo
+	for _, dir := range topLevelDirs {
+		if dir.Name() != ".git" {
+			topLevelDirsNoDotGit = append(topLevelDirsNoDotGit, dir)
+			break
+		}
+	}
+	if err != nil {
+		return "", err
+	}
+	numDirs := len(topLevelDirsNoDotGit)
+	if numDirs != 1 {
+		return "", err
+	}
+	topLevelDir := topLevelDirsNoDotGit[0]
+	if topLevelDir.Name() != "environments" {
+		return "", err
```

Comment:
Confusing return here - can we not return a message that says, 'There must be exactly one directory inside /environments.' 

Should .git not be in the root directory, not at /environments/.git? I'm a bit confused.

Reply:
Yeah good suggestion, added

It's not clear from the code but at that point, we are in the root directory, which is where they may be a .git folder - we're not in /environments at this point

Reply:
Also fixed up the thing you mentioned - now you don't need just one folder at the top level called environments